Added health check docs, & details on updating apps (#11729)
docker/docker.github.io: engine/context/aci-integration.md

>
> The current Docker Azure integration does not allow fetching a combined log stream from all the containers that make up the Compose application.

## Updating applications

> From a deployed Compose application, you can update the application by re-deploying it with the same ACI group name: `docker compose up --project-name` with the Compose project name already used in previous deployments.

Reviewer: ```suggestion
From a deployed Compose application, you can update the application by re-deploying it with the same project name: `docker compose up --project-name PROJECT`.

```

> Updating an application means the ACI node will be reused, and the application will keep the IP address already allocated to expose ports, if any. ACI also has some limitations on what can be updated in an existing application (you will not be able to change CPU/memory reservation for example), in these cases you need to deploy a new application from scratch.

Reviewer: ```suggestion
Updating an application means the ACI node will be reused, and the application will keep the same IP address that was previously allocated to expose ports, if any. ACI has some limitations on what can be updated in an existing application (you will not be able to change CPU/memory reservation for example), in these cases you need to deploy a new application from scratch.

```

> Updating is the default behaviour if you invoke several times `docker compose up` on the same compose file, since the Compose project name is derived from the compose file location by default. You need to explicitely execute `docker compose down` before running again `docker compose up` in order to totally reset a Compose application.

Reviewer: ```suggestion
Updating is the default behavior if you invoke `docker compose up` on the an already deployed Compose file, since the Compose project name is derived from the directory where the Compose file is located by default. You need to explicitly execute `docker compose down` before running `docker compose up` again in order to totally reset a Compose application.

```

## Releasing resources

> Single containers and Compose applications can be removed from ACI with

Reviewer: Can't change this as a suggestion unfortunately but would be good to rephrase so that we say it's only stopped/terminated containers that are cleaned up by prune.

Author: It's actually any state that is not "running". ACI containers sometimes ends up in weird states that we can't all list. 
`The `docker prune` command removes deployments that are not currently running. ` is exactly what happens.

Reviewer: OK then SGTM

When using `docker run`, you can pass the environment variables to ACI containers using the `--env` flag.
For Compose applications, you can specify the environment variables in the Compose file with the `environment` or `env-file` service field, or with the `--environment` command line flag.

## Healthchecks

> You can specify container healthchecks either with `docker run --healthcheck-xxx` flags, or in a compose file with the healthcheck section.

Reviewer: ```suggestion
You can specify a container health checks using either the `--healthcheck-` prefixed. flags with `docker run`, or in a Compose file with the healthcheck section of the service.

```

> Healthchecks are converted to ACI LivenessProbe. ACI will run the healthcheck command periodically, and when it fails, the container will be terminated.

Reviewer: ```suggestion
Health checks are converted to ACI `LivenessProbe`s. ACI will run the health check command periodically, and if it fails, the container will be terminated.

```

> Healthchecks must be used in addition to restart policies, to ensure the container is then restarted ; using no restart policy will lead to containers being killed but not restarted by healthchecks.

Reviewer: ```suggestion
Health checks must be used in addition to restart policies, to ensure the container is then restarted on termination. The default restart policy for `docker run` is `no` which will not restart the container. The default restart policy for Compose is `any` which will always try restarting the service containers.

```

Example using `docker run`:

```console
docker --context acicontext run -p 80:80 --restart always --health-cmd "curl http://localhost:80" --health-interval 3s  nginx
```

> Example using compose files:

Reviewer 2: ```suggestion
Example using Compose files:

```
